javaslang/src/main/java/javaslang/$.java

```diff
@@ -33,8 +33,7 @@
     // -- javaslang.collection
 
     // List
-    @Unapply static <T> Tuple2<T, List<T>> List(List.Cons<T> cons) { return Tuple.of(cons.head(), cons.tail()); }
-    @Unapply static <T> Tuple0 List(List.Nil<T> nil) { return Tuple.empty(); }
+    @Unapply static <T> Tuple2<T, List<T>> List(List<T> cons) { return Tuple.of(cons.head(), cons.tail()); }
```

Comment:
@danieldietrich, this wasn't tested so I presumed it wasn't working before either :p